tests/plugin/test_http_proxy_plugins_with_tls_interception.py

```diff
@@ -178,32 +176,59 @@ def send(raw: bytes) -> int:
         )
         self.assertFalse(self.protocol_handler.work.has_buffer())
 
-    def test_modify_post_data_plugin(self) -> None:
+        #
         original = b'{"key": "value"}'
         modified = b'{"key": "modified"}'
         self.client_ssl_connection.recv.return_value = build_http_request(
             b'POST', b'/',
             headers={
                 b'Host': b'uni.corn',
-                b'Content-Type': b'application/x-www-form-urlencoded',
                 b'Content-Length': bytes_(len(original)),
+                b'Content-Type': b'application/x-www-form-urlencoded',
             },
             body=original,
         )
-        self.protocol_handler._run_once()
-        self.server.queue.assert_called_with(
-            build_http_request(
-                b'POST', b'/',
-                headers={
-                    b'Host': b'uni.corn',
-                    b'Content-Length': bytes_(len(modified)),
-                    b'Content-Type': b'application/json',
-                },
-                body=modified,
-            ),
+        await self.protocol_handler._run_once()
+        self.server.queue.assert_called_once()
+        # pkt = build_http_request(
+        #     b'POST', b'/',
+        #     headers={
+        #         b'Host': b'uni.corn',
+        #         b'Content-Length': bytes_(len(modified)),
+        #         b'Content-Type': b'application/json',
+        #     },
+        #     body=modified,
+        # )
+        response = HttpParser.response(
+            self.server.queue.call_args_list[0][0][0].tobytes(),
         )
+        self.assertEqual(response.body, modified)
+
+    @pytest.mark.asyncio    # type: ignore[misc]
+    @pytest.mark.parametrize(
+        '_setUp',
+        [('test_man_in_the_middle_plugin')],
+        indirect=True,
+    )   # type: ignore[misc]
+    async def test_man_in_the_middle_plugin(self) -> None:
+        await self.protocol_handler._run_once()
+
+        self.assertEqual(self.mock_sign_csr.call_count, 1)
```

Comment:
Having the actual assertion in a different place will probably make the pytest output worse (it patches the bytecode of the `assert` statements to retrieve more context about the local variables in the place where it's present)

Reply:
I agree, it's an anti-pattern within pytest eco-system.  Lemme deprecate `Assertions` too within this PR :D